app/uk/gov/hmrc/breathingspaceifproxy/connector/service/UpstreamConnector.scala

```diff
@@ -77,20 +77,20 @@ trait DownstreamConnector extends HttpErrorFunctions with Logging with UsingCirc
   ): ResponseValidation[T] =
     statusCode match {
       case NOT_FOUND => notFound(message)
-      case FORBIDDEN => logErrorAndGenUpstreamResponse(FORBIDDEN, message, BREATHING_SPACE_EXPIRED)
-      case CONFLICT => logErrorAndGenUpstreamResponse(CONFLICT, message, CONFLICTING_REQUEST)
-      case _ => logErrorAndGenUpstreamResponse(statusCode, message, SERVER_ERROR)
+      case FORBIDDEN => logWarningAndGenDownstreamResponse(FORBIDDEN, message, BREATHING_SPACE_EXPIRED)
+      case CONFLICT => logInfoAndGenDownstreamResponse(CONFLICT, message, CONFLICTING_REQUEST)
+      case _ => logWarningAndGenDownstreamResponse(statusCode, message, SERVER_ERROR)
     }
 
   private def handleUpstream5xxError[T](statusCode: Int, message: String)(
     implicit r: RequestId
   ): ResponseValidation[T] =
     statusCode match {
-      case BAD_GATEWAY => logErrorAndGenUpstreamResponse(BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
+      case BAD_GATEWAY => logErrorAndGenDownstreamResponse(BAD_GATEWAY, message, UPSTREAM_BAD_GATEWAY)
       case SERVICE_UNAVAILABLE =>
-        logErrorAndGenUpstreamResponse(SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
-      case GATEWAY_TIMEOUT => logErrorAndGenUpstreamResponse(GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
-      case _ => logErrorAndGenUpstreamResponse(statusCode, message, SERVER_ERROR)
+        logErrorAndGenDownstreamResponse(SERVICE_UNAVAILABLE, message, UPSTREAM_SERVICE_UNAVAILABLE)
+      case GATEWAY_TIMEOUT => logErrorAndGenDownstreamResponse(GATEWAY_TIMEOUT, message, UPSTREAM_TIMEOUT)
+      case _ => logErrorAndGenDownstreamResponse(statusCode, message, SERVER_ERROR)
```

Comment:
I think all this logic could be simplified and centralised inside a general "logAndResponse" method. Atm we are just passing the statusCode and message down the stack, the only thing that changes here is the log level and the response sent to PEGA. Could this not be controled in one "logAndResponse" method matching status code to log level and response?

Reply:
I would prefer to keep it this way. tbh I do not like to much to have a method with a big `match`. This way the logic between 4xx and 5xx is at least separated.